vcd/resource_vcd_lb_virtual_server.go

```diff
@@ -0,0 +1,281 @@
+package vcd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func resourceVcdLBVirtualServer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdLBVirtualServerCreate,
+		Read:   resourceVcdLBVirtualServerRead,
+		Update: resourceVcdLBVirtualServerUpdate,
+		Delete: resourceVcdLBVirtualServerDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceVcdLBVirtualServerImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "vCD organization in which the LB Virtual Server is located",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "vCD virtual datacenter in which the LB Virtual Server is located",
+			},
+			"edge_gateway": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway name in which the LB Virtual Server is located",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Unique Virtual Server name",
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Virtual Server description",
+			},
+			"enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Default:     true,
+				Optional:    true,
+				Description: "Defines if the virtual server is enabled",
+			},
+			"ip_address": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "IP address that the load balancer listens on",
+				ValidateFunc: validation.SingleIP(),
+			},
+			"protocol": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Protocol that the virtual server accepts",
+				ValidateFunc: validateCase("lower"),
+			},
+			"port": &schema.Schema{
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "Port number that the load balancer listens on",
+			},
+			"enable_acceleration": &schema.Schema{
```

Comment:
Why is the default  for this property `false`, while we know that without acceleration the LB won't work?

Reply:
It's not necessarily not going to work. The idea behind it is such as per [vCD doc](https://docs.vmware.com/en/vCloud-Director/9.0/com.vmware.vcloud.tenantportal.doc/GUID-7E082E77-B459-4CE7-806D-2769F7CB5624.html). A asked for it in the test so that it is less steps. The UI has it disabled by default:
> Click the Acceleration Enabled icon to configure the load balancer to use the faster L4 engine rather than L7 engine. The L4 TCP VIP is processed before the edge gateway firewall so no Allow firewall rule is required. 

> L7 VIPs for HTTP and HTTPS are processed after the firewall, so when Acceleration Enabled is not selected, an edge gateway firewall rule must exist to allow access to the L7 VIP for those protocols. When Acceleration Enabled is selected and the server pool is in non-transparent mode, an SNAT rule is added, so you must ensure that the firewall is enabled on the edge gateway.